src/Domain/Structure/ElementId.cpp

```diff
@@ -100,6 +105,31 @@ ElementId<VolumeDim> ElementId<VolumeDim>::id_of_parent(const size_t dim) const
   return {block_id(), new_segment_ids, grid_index()};
 }
 
+template <size_t VolumeDim>
```

Comment:
we want these inlined so please move them back

Reply:
Do they inline automatically somehow? They are not marked `inline`

Reply:
Any function defined inside the class body is implicitly marked as `inline`, but that's a different `inline` than what I mean. The compiler can elide a function call if it knows the function definition because it is in a header file. For short functions like getters, it's common to keep them in the header file to avoid the function call overhead